src/Neo/SmartContract/Native/NeoToken.cs

```diff
@@ -83,6 +82,15 @@ public override BigInteger TotalSupply(IReadOnlyStoreView snapshot)
             return TotalAmount;
         }
 
+        protected override bool StateIsClean(ApplicationEngine engine, NeoAccountState state)
+        {
+            if (engine.IsHardforkEnabled(Hardfork.HF_Echidna))
+            {
+                if (state.VoteTo != null) return false;
```

Comment:
```suggestion
                if (state.VoteTo != null)
                {
                  throw new UncatcheableException("You need to vote for null if you want to transfer all your neos");
                }
```

something like this

Reply:
Neat, but then people will complain that "Neo forces us to make two transactions instead of one".